src/network/networkd-neighbor.c

```diff
@@ -213,6 +213,28 @@ static bool neighbor_equal(const Neighbor *n1, const Neighbor *n2) {
         return neighbor_compare_func(n1, n2) == 0;
 }
 
+static void log_neighbor_debug(const Neighbor *neighbor, const char *str, const Link *link) {
+        assert(neighbor);
+        assert(str);
+
+        if (DEBUG_LOGGING) {
```

Comment:
I'd always invert the check here, and return here early, in order to keep indentation levels low. i.e.:

```c
if (!DEBUG_LOGGING)
        return;
```

but doesn't matter